daemon/containerd/service.go

```diff
@@ -0,0 +1,245 @@
+package containerd
+
+import (
+	"context"
+	"io"
+
+	"github.com/containerd/containerd"
+	"github.com/containerd/containerd/platforms"
+	"github.com/docker/distribution"
+	"github.com/docker/distribution/reference"
+	"github.com/docker/docker/api/types"
+	"github.com/docker/docker/api/types/backend"
+	"github.com/docker/docker/api/types/filters"
+	imagetype "github.com/docker/docker/api/types/image"
+	registrytypes "github.com/docker/docker/api/types/registry"
+	"github.com/docker/docker/builder"
+	"github.com/docker/docker/container"
+	"github.com/docker/docker/daemon"
+	"github.com/docker/docker/daemon/images"
+	"github.com/docker/docker/errdefs"
+	"github.com/docker/docker/image"
+	"github.com/docker/docker/layer"
+	"github.com/opencontainers/go-digest"
+	ocispec "github.com/opencontainers/image-spec/specs-go/v1"
+)
+
+type containerdStore struct {
+	client *containerd.Client
+}
+
+func NewService(c *containerd.Client) daemon.ImageService {
```

Comment:
@rumpl I changed this to return the interface to make the Linter happy; alternatively, we could export ContainerdStore (but that diff would be larger, so harder for you to do a rebase); let me know which one you prefer

Reply:
Ah, crap, that doesn't work of course;

```
#41 0.553 package github.com/docker/docker/cmd/dockerd
#41 0.553 	imports github.com/docker/docker/daemon
#41 0.553 	imports github.com/docker/docker/daemon/containerd
#41 0.553 	imports github.com/docker/docker/daemon: import cycle not allowed